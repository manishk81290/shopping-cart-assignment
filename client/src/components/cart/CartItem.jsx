import React from "react";
import { Row, Col, Image, Button } from "react-bootstrap";

export const CartItem = ({ item, index, updateQuantity }) => {
  return (
    <Row xs={12} className="cart-item">
      <Col xs={3}>
        <Image src={`${process.env.PUBLIC_URL}${item.imageURL}`} thumbnail />
      </Col>
      <Col xs={9} className="pl-0">
        <h5>{item.name}</h5>
        <Button
          variant="primary"
          className="theme-button"
          onClick={() => {
            updateQuantity("decrement", index);
          }}
        >
          -
        </Button>
        <span>{item.count}</span>
        <Button
          variant="primary"
          className="theme-button"
          onClick={() => {
            updateQuantity("increment", index);
          }}
        >
          +
        </Button>
        <span>x</span>
        <span>Rs.{item.price}</span>
        <span className="total">Rs.{item.price * item.count}</span>
      </Col>
    </Row>
  );
};

export default CartItem;
