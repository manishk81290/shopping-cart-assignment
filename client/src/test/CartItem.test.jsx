import React from "react";
import CartItem from "./../components/cart/CartItem";
import ShallowRenderer from "react-test-renderer/shallow";

const testProps = {
  item: {
    name: "Red Label Tea, 500 gm Carton",
    imageURL: "/static/images/products/beverages/red-label.jpg",
    description:
      "Brooke Bond Red Label is one of Indias largest selling packaged tea brands. Brooke Bond Red Label Tea is a blend CTC tea with best quality leaves.",
    price: 205,
    stock: 50,
    category: "5b675e5e5936635728f9fc30",
    sku: "bev-red-label-500",
    id: "5b6c6f8301a7c3842953088d",
    count: 3,
  },
  index: 1,
  updateQuantity: 1,
};

it("CartItem with mock useContext hook", () => {
  const element = new ShallowRenderer().render(<CartItem {...testProps} />);
  expect(element).toBeTruthy();
  expect(element).toMatchSnapshot();
});
