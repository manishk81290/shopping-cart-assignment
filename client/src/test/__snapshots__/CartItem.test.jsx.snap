// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem with mock useContext hook 1`] = `
<Row
  className="cart-item"
  noGutters={false}
  xs={12}
>
  <Col
    xs={3}
  >
    <Image
      fluid={false}
      rounded={false}
      roundedCircle={false}
      src="/static/images/products/beverages/red-label.jpg"
      thumbnail={true}
    />
  </Col>
  <Col
    className="pl-0"
    xs={9}
  >
    <h5>
      Red Label Tea, 500 gm Carton
    </h5>
    <Button
      active={false}
      className="theme-button"
      disabled={false}
      onClick={[Function]}
      variant="primary"
    >
      -
    </Button>
    <span>
      3
    </span>
    <Button
      active={false}
      className="theme-button"
      disabled={false}
      onClick={[Function]}
      variant="primary"
    >
      +
    </Button>
    <span>
      x
    </span>
    <span>
      Rs.
      205
    </span>
    <span
      className="total"
    >
      Rs.
      615
    </span>
  </Col>
</Row>
`;
